### ISSUE TITLE
Running vitastor-disk prepare results in osds are full warning

### ISSUE THREAD

**Issue by joeplaa** [vitalif/vitastor#73]
I'm testing Vitastor as an alternative to Linstor and Ceph. So far this hasn't been as smooth a ride as I hoped.

When running `vitastor-disk prepare /dev/sda --disable_data_fsync off` on my three nodes I get:

```shellsession
root@jpl-proxmox7:~# vitastor-cli status
  cluster:
    etcd: 3 / 3 up, 36 K database size
    mon:  3 up, master jpl-proxmox7
    osd:  0 / 3 up

  warning:
    3 osds are full
  
  data:
    raw:   0 B used, 0 B / 0 B available
    state: 0 B clean
    pools: 0 / 0 active
    pgs:   
  
  io:
    client: 0 B/s rd, 0 op/s rd, 0 B/s wr, 0 op/s wr
```

I found this when reading the superblock: `"data_size": 0`. Does this mean no size? Or is this unrelated? What am I missing?

```json
{
    "autosync_writes": "512",
    "bitmap_granularity": 4096,
    "block_size": 131072,
    "data_device": "/dev/disk/by-partuuid/0ab54fa0-bfa3-f248-99a2-73b9d9540e86",
    "data_offset": 203382784,
    "data_size": 0,
    "disable_data_fsync": false,
    "disable_device_lock": false,
    "disable_journal_fsync": false,
    "disable_meta_fsync": false,
    "disk_alignment": 4096,
    "immediate_commit": "none",
    "journal_block_size": 4096,
    "journal_device": "",
    "journal_no_same_sector_overwrites": true,
    "journal_offset": 4096,
    "journal_sector_buffer_count": 1024,
    "max_flusher_count": "256",
    "meta_block_size": 4096,
    "meta_device": "",
    "meta_offset": 134221824,
    "min_flusher_count": "32",
    "osd_num": 1,
    "skip_cache_check": false
}
```

**Comment by vitalif** [comment]
Hi, check osd logs in journalctl, it seems they didn't start or weren't started. "Full" osds in your case are just osds which never reported any statistics because they never started. Data_size=0 is ok, it means auto-guess.

**Comment by vitalif** [comment]
P.S. just committed a fix so that vitastor-cli status doesn't report new osds as "full" anymore :-)

**Comment by joeplaa** [comment]
Thanks for the quick response. This is from the logs:
```
Nov 17 09:22:31 jpl-proxmox7 vitastor-disk[6661]: Reading blockstore metadata
Nov 17 09:22:31 jpl-proxmox7 vitastor-disk[6661]: Metadata is corrupt or too old (pre-0.6.x).
Nov 17 09:22:31 jpl-proxmox7 vitastor-disk[6661]:  If this is a new OSD, please zero out the metadata area before starting it.
Nov 17 09:22:31 jpl-proxmox7 vitastor-disk[6661]:  If you need to upgrade from 0.5.x, convert metadata with vitastor-disk.
```

**Comment by vitalif** [comment]
Oh, sorry. This is a known bug already fixed after 1.9.3 (I'll release it soon) - vitastor-disk prepare didn't correctly zero OSD metadata in cases when of really small OSDs, like 50 GB partitions. Please zero the beginning of your test disk manually, purge the device (vitastor-disk purge) and prepare it again... :-)

**Comment by joeplaa** [comment]
Had to wipe the disks, `dd` them with the command you provided above and then run the disk-prepare again. Now I indeed have 3 osd's up! Thanks.

```shellsession
joep@jpl-proxmox7:~$ vitastor-cli status
  cluster:
    etcd: 3 / 3 up, 64 K database size
    mon:  3 up, master jpl-proxmox9
    osd:  3 / 3 up
  
  data:
    raw:   0 B used, 698.1 G / 698.1 G available
    state: 0 B clean
    pools: 0 / 0 active
    pgs:   
  
  io:
    client: 0 B/s rd, 0 op/s rd, 0 B/s wr, 0 op/s wr
```
